frontend/src/firebase/actions.ts

```diff
@@ -182,6 +170,39 @@ export const editTaskWithDiff = (
     });
   }
   batch.set(tasksCollection().doc(taskId), mainTaskEdits, { merge: true });
+  batch.commit();
+};
+
+export const editTaskWithDiff = (
+  taskId: string,
+  editType: EditType,
+  { mainTaskEdits, subTaskCreations, subTaskEdits, subTaskDeletions }: Diff,
+): void => {
+  const batch = db().batch();
+  if (editType === 'FORKING_MASTER_TEMPLATE') {
+    (async () => {
+      const { tasks } = store.getState();
+      const repeatingTaskMaster = tasks.get(taskId) as RepeatingTask;
+      const { id, ...taskWithoutID } = repeatingTaskMaster;
+      const newTaskId = getNewTaskId();
+      batch.set(tasksCollection().doc(newTaskId), { newTaskId, ...taskWithoutID }, { merge: true });
```

Comment:
this spread is not safe. `...taskWithoutID` also contains `type: 'MASTER_TEMPLATE', repeats: ...`. You need to exclude them and then add `type: 'ONE_TIME'` for the new task.
It's advised that you write something like 
```typescript
const newOneTimeTask: FirestoreOneTimeTask = {...}
```
to ensure you indeed have all the required stuff and nothing more.

Reply:
Other than that, things are looking good.